File: smartmoneyserver/src/main/java/fpt/aptech/server/dto/event/EventResponse.java
package fpt.aptech.server.dto.event;

import lombok.Builder;

import java.math.BigDecimal;
import java.time.LocalDate;

/**
 * DTO để trả về thông tin chi tiết của một sự kiện.
 * Bao gồm các trường tính toán sẵn để Frontend dễ dàng hiển thị.
 */
@Builder(toBuilder = true) // Bật tính năng toBuilder()
public record EventResponse(
        Integer id,
        String eventName,
        String eventIconUrl,
        LocalDate beginDate,
        LocalDate endDate,
        Boolean finished,
        String currencyCode,

        // Các trường tính toán thêm
        BigDecimal totalIncome,
        BigDecimal totalExpense,
        BigDecimal netAmount // = totalIncome - totalExpense
) {}

File: APP MOBILE/smart_money/lib/modules/event/models/event_response.dart
/// Response hiển thị thông tin sự kiện.
/// Tương ứng: EventResponse.java (server)
class EventResponse {
  final int id;
  final String eventName;
  final String? eventIconUrl;
  final DateTime? beginDate;
  final DateTime endDate;
  final bool? finished;
  final String? currencyCode;
  final double totalIncome;
  final double totalExpense;
  final double netAmount;

  const EventResponse({
    required this.id,
    required this.eventName,
    this.eventIconUrl,
    this.beginDate,
    required this.endDate,
    this.finished,
    this.currencyCode,
    this.totalIncome = 0,
    this.totalExpense = 0,
    this.netAmount = 0,
  });

  factory EventResponse.fromJson(Map<String, dynamic> json) {
    return EventResponse(
      id: json['id'] as int,
      eventName: json['eventName'] as String,
      eventIconUrl: json['eventIconUrl'] as String?,
      beginDate: json['beginDate'] != null
          ? DateTime.parse(json['beginDate'] as String)
          : null,
      endDate: DateTime.parse(json['endDate'] as String),
      finished: json['finished'] as bool?,
      currencyCode: json['currencyCode'] as String?,
      totalIncome: (json['totalIncome'] as num?)?.toDouble() ?? 0,
      totalExpense: (json['totalExpense'] as num?)?.toDouble() ?? 0,
      netAmount: (json['netAmount'] as num?)?.toDouble() ?? 0,
    );
  }
}

